Description: Use qemu:///system as the default URI
 We've done this in Ubuntu for ages, and can't switch it now.  However
 the old patch ended up changing the behavior of various ways of
 specifying URI preferences.  We don't want that - we only want to
 change the default if nothing is specified.
Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Forwarded: no

Index: libvirt/src/libvirt.c
===================================================================
--- libvirt.orig/src/libvirt.c	2016-08-15 14:44:03.426494533 +0200
+++ libvirt/src/libvirt.c	2016-08-15 14:50:58.390483432 +0200
@@ -907,15 +907,16 @@ virConnectGetDefaultURI(virConfPtr conf,
 {
     int ret = -1;
     const char *defname = virGetEnvBlockSUID("LIBVIRT_DEFAULT_URI");
+    const char *fallback = "qemu:///system";
     if (defname && *defname) {
         VIR_DEBUG("Using LIBVIRT_DEFAULT_URI '%s'", defname);
         if (VIR_STRDUP(*name, defname) < 0)
             goto cleanup;
     } else {
-        if (virConfGetValueString(conf, "uri_default", name) < 0)
-            goto cleanup;
-
-        if (*name)
+        if (virConfGetValueString(conf, "uri_default", name) < 0) {
+            VIR_DEBUG("Falling back to uri qemu:///system");
+            *name = fallback;
+        } else if (*name)
             VIR_DEBUG("Using config file uri '%s'", *name);
     }
 
